src/app/new-courses-page/course-duration/course-duration.component.ts

```diff
@@ -1,16 +1,92 @@
-import { ChangeDetectionStrategy, Component, EventEmitter, Input, Output } from '@angular/core';
+import { 
+  ChangeDetectionStrategy,
+  Component,
+  forwardRef,
+  Input
+} from '@angular/core';
+import {
+  Validators,
+  ControlValueAccessor,
+  NG_VALUE_ACCESSOR,
+  AbstractControl,
+  ValidationErrors,
+  NG_VALIDATORS,
+  Validator,
+  FormControl
+ } from '@angular/forms';
+
+import { Subject } from 'rxjs';
+import { takeUntil } from 'rxjs/operators';
+
+export const SELECT_CONTROL_VALUE_ACCESSOR: any = {
```

Comment:
avoid any, try not to use that. There is no value in typescript if you use any
The correct type would be
`any -> ExistingProvider`